src/averageAges.js

```diff
@@ -55,7 +67,17 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const arrDiffAge = (!onlyWithSon)
+    ? people.filter(child =>
+      people.find(mother => child.mother === mother.name))
+    : people.filter(child =>
+      people.find(mother => child.mother === mother.name) && child.sex === 'm');
+
+  const averageAgeDiff = arrDiffAge.reduce((count, child) =>
+    count + (child.born - people.find(mother =>
+      child.mother === mother.name).born), 0) / arrDiffAge.length;
```

Comment:
Better create variable `motherBorn` for example, this way code would be easier to read.